screens/PokemonList.tsx

```diff
@@ -1,28 +1,20 @@
 import React, { useEffect, useState } from 'react';
 import { StyleSheet, Text, View, FlatList, Dimensions } from 'react-native';
 import { getAllPokemons } from '../api/pokemon';
+import { Pokemon } from '../api/pokemon';
 
 const keyExtractor = ({ name }: { name: string }): string => name;
 
 export default function PokemonList() {
-  interface Pokemon {
-    name: string;
-    weight: number;
-    height: number;
-    base_experience: number;
-    sprites: [object];
-    url: string;
-  }
-
   const step = 20;
   const [batchStart, setBatchStart] = useState(0);
   const [loading, setLoading] = useState(true);
-  const [pokemons, setPokemons] = useState([]);
+  const [pokemons, setPokemons] = useState<ReadonlyArray<Pokemon> | null>([]);
```

Comment:
I think you don't need `| null` type now -> you just cannot `setPokemons(allPokemons);` (it can be null in case of an error), but `if (allPokemons) setPokemons(allPokemons)`. or `setPokemons([...(allPokemons ?? [])])` -> it's better to always keep pokemons as array

You will be able to remove `if (pokemons)` below then

Reply:
btw there are very nice operators in Typescript:
https://www.infoq.com/news/2019/11/typescript-3-7-release/